fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -1822,17 +1765,64 @@ private void analyzePartitionExprs() {
                     if (!refBaseTablePartitionExprs.containsKey(refBaseTable)) {
                         continue;
                     }
-                    Expr partitionExpr = refBaseTablePartitionExprs.get(refBaseTable);
+                    List<Expr> partitionExprs = refBaseTablePartitionExprs.get(refBaseTable);
                     TableName tableName = new TableName(baseTableInfo.getCatalogName(),
                             baseTableInfo.getDbName(), baseTableInfo.getTableName());
-                    analyzePartitionExpr(connectContext, refBaseTable, tableName, partitionExpr);
+                    for (Expr partitionExpr : partitionExprs) {
+                        analyzePartitionExpr(connectContext, refBaseTable, tableName, partitionExpr);
+                    }
                 }
             }
+            // analyze partition slots for ref base tables
+            analyzeRefBaseTablePartitionSlots();
+            // analyze partition columns for ref base tables
+            analyzeRefBaseTablePartitionColumns();
         } catch (Exception e) {
             LOG.warn("Analyze partition exprs failed", e);
         }
     }
 
+    private void analyzeRefBaseTablePartitionExprs() {
+        Map<Table, List<Expr>> refBaseTablePartitionExprMap = Maps.newHashMap();
+        for (BaseTableInfo tableInfo : baseTableInfos) {
+            Table table = MvUtils.getTableChecked(tableInfo);
+            List<MVPartitionExpr> mvPartitionExprs = MvUtils.getMvPartitionExpr(partitionExprMaps, table);
+            if (CollectionUtils.isEmpty(mvPartitionExprs)) {
+                LOG.info("Base table {} contains no partition expr, skip", table.getName());
```

Comment:
would this log be printed each time that mv is refreshed or rewritten?

Reply:
No. `analyzeRefBaseTablePartitionExprs` only called in the `onReload` phase which should be called in FE Restart.